writing-gleam/command-line-reference.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+
+# This script uses the gleam compiler CLI to generate markdown
+# from its own help output.
+
+set -e
+
+trim() {
+  sed 's/^[[:space:]]*//' | sed 's/[[:space:]]*$//'
+}
+
+find_description() {
+  sed -n '2,/^USAGE:/p' | grep -v 'USAGE:' | trim
+}
+
+find_subcommands() {
+  sed -n '/SUBCOMMANDS:/,/^$/p' | grep -v 'SUBCOMMANDS:' | grep -v '^$' | trim | cut -d' ' -f1
+}
+
+find_usage() {
+  sed -n '/USAGE:/,/^$/p' | grep -v 'USAGE:' | grep -v '^$' | trim
+}
+
+find_options() {
+  # We ignore the --help option as it is not useful in these docs,
+  # grep returns 1 if no matches, we need to ignore that for the pipeline to work.
+  set +e
+  # Some subcommands empty lines in the middle of the options lists, so we cannot use `sed -n '/OPTIONS:/,/^$/p'`.
+  # There are some options that span multiple lines, we need to join them.
+  sed -n '/OPTIONS:/,//p' | grep -v 'OPTIONS:' | tr '\n' '§' | sed -E 's/§ *-/\n-/g' | sed 's/§ */ /g' | trim | grep -v -- '--help'
+  set -e
+}
+
+# Render markdown help for a subcommand, or a subcommand under it.
+show_docs() {
+    subcommand="$1"
```

Comment:
Could we have two space indentation in this function please?

Reply:
definitely , (and I should look into getting a shell formatter for my editor)

Reply:
If you find a good one let me know!